### PR TITLE
rdma-core: add v52.0, enable all modules

### PR DESCRIPTION
**Pull request by valgur**
### Summary
Changes to recipe:  **rdma-core/52.0**

#### Motivation
#18979 requires `build_librdmacm=True`. Enabled the remaining `build_libmlx5` as well, since it does not add any dependencies or other notable overhead.

#### Details
https://github.com/linux-rdma/rdma-core/releases/tag/v52.0

---
- [ ] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [ ] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by valgur** [comment]
@perseoGI

**Comment by perseoGI** [comment]
There is a compilation issue with `librdmacm` in conan 1 pipeline. Could you take a look?
I'm agree to enable the remaining options by default, but, couldn't that be done from `libfabrick`?

**Comment by valgur** [comment]
> couldn't that be done from libfabrick?

Not on CCI itself, since overriding an option would require the dependency to be rebuilt and the CI does not use `--build-missing`.

**Review by perseoGI — APPROVED** [review]
LGTM now, thanks! And thanks for the clarification

File: recipes/rdma-core/all/conanfile.py [hunk 1]
```diff
@@ -73,6 +72,10 @@ def generate(self):
         # tc.variables["ENABLE_STATIC"] = not self.options.shared
         tc.variables["NO_PYVERBS"] = True
         tc.variables["NO_MAN_PAGES"] = True
+        # Otherwise get set to ${install_prefix}/car/run and the paths in
+        # https://github.com/linux-rdma/rdma-core/blob/v52.0/buildlib/config.h.in
+        # can exceed the 108-character limit for socket paths on Linux
+        tc.variables["CMAKE_INSTALL_RUNDIR"] = "/var/run"
```
Comment (uilianries) [inline recipes/rdma-core/all/conanfile.py:78]:
```suggestion
        if "CMAKE_INSTALL_RUNDIR" not in self.conf.get("tools.cmake.cmaketoolchain:extra_variables", check_type=dict, default={}):
            tc.variables["CMAKE_INSTALL_RUNDIR"] = self."/var/run"
```

I would keep it more flexible as we have issues when declaring hardcoded system path in a recipe.

Comment (valgur) [reply recipes/rdma-core/all/conanfile.py]: Thanks! That's a much better solution.

**Pull request merged**